clippy_lints/src/operators/mod.rs

```diff
@@ -299,6 +300,43 @@ declare_clippy_lint! {
     "unnecessary double comparisons that can be simplified"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for double comparisons that can never succeed
+    ///
+    /// ### Why is this bad?
+    /// The whole expression can be replaced by `false`,
+    /// which is probably not the programmer's intention
+    ///
+    /// ### Example
+    /// ```rust
+    /// status_code <= 400 && status_code > 500;
+    /// ```
+    #[clippy::version = "1.71.0"]
+    pub IMPOSSIBLE_DOUBLE_CONST_COMPARISONS,
```

Comment:
issue: unsure about the name. opened a thread https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/lint.20naming.3A.20double.20comparisons

i think IMPOSSIBLE_DOUBLE_COMPARISONS without the `CONST` is slightly better

Reply:
from chat

> Maybe something like `redundant_comparisons` + `impossible_comparisons`

> The `double`/`const` part seem more like implementation details to me. Also `double` made me think of `f64` initially


we should probably make these work for all numbers and rename the lints